pkg/boot/mudp/multicaster.go

```diff
@@ -0,0 +1,64 @@
+package mudp
+
+import (
+	"log"
+	"net"
+	"strings"
+)
+
+const (
+	maxDatagramSize = 8192
+)
+
+type multicaster struct {
+	addr  *net.UDPAddr
+	close chan chan error
+	conn  *net.UDPConn
+}
+
+func NewMulticaster(addr string) (mc *multicaster, err error) {
+	udpAddr, err := net.ResolveUDPAddr("udp4", addr)
+	if err != nil {
+		return
+	}
+	return &multicaster{addr: udpAddr, close: make(chan chan error)}, nil
+}
+
+func (mc *multicaster) Listen(ready chan bool, handler func(*net.UDPAddr, int, []byte)) {
+	conn, err := net.ListenMulticastUDP("udp4", nil, mc.addr)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
This will cause the entire program to terminate, whereas we should allow the caller to handle the error.

I suggest instantiating `conn` from the body of `NewMudp` and then passing it into `Listen`.  This also eliminates the need for the `ready` channel.